category/category.go

```diff
@@ -0,0 +1,255 @@
+	db := m.db.Create(relationship)
+	if db.Error != nil {
+		if gormsupport.IsUniqueViolation(db.Error, "workitemtype_categories_idx") {
+			return errors.NewBadParameterError("category+workitemtype", relationship.CategoryID).Expected("unique")
+		}
+		return errors.NewInternalError(db.Error.Error())
+	}
+	return nil
+}
+
+// Create creates category. This function is used to populate categories table during migration -> PopulateCategories()
+func (m *GormRepository) Create(ctx context.Context, category *Category) (*Category, error) {
+	if strings.TrimSpace(category.Name) == "" {
+		return nil, errors.NewBadParameterError("Name", category.Name)
+
+	}
+	db := m.db.Create(category)
+	if db.Error != nil {
+		if gormsupport.IsUniqueViolation(db.Error, "categories_name_idx") {
+			return nil, errors.NewBadParameterError("Name", category.Name).Expected("unique")
+		}
+		return nil, errors.NewInternalError(db.Error.Error())
+	}
+	log.Info(ctx, map[string]interface{}{
+		"category_id": category.ID,
+	}, "Category created successfully")
+	return category, nil
+}
+
+// LoadAllRelationshipsOfCategory loads all the relationships of a category. This is required for workitemtype filtering.
+func (m *GormRepository) LoadAllRelationshipsOfCategory(ctx context.Context, categoryID uuid.UUID) ([]*WorkItemTypeCategoryRelationship, error) {
+	// Check if category is present
+	getCategory := Category{}
+	db := m.db.Model(&getCategory).Where("id=?", categoryID).Find(&getCategory)
+	if db.RecordNotFound() {
+		log.Error(ctx, map[string]interface{}{
+			"categories_id": categoryID,
+		}, "category not found")
+		return nil, errors.NewNotFoundError("category", categoryID.String())
+	}
+	if err := db.Error; err != nil {
+		return nil, errors.NewInternalError(err.Error())
+	}
+	relationship := []*WorkItemTypeCategoryRelationship{}
+	db = m.db.Model(&relationship).Where("category_id=?", categoryID).Find(&relationship)
+	if db.RecordNotFound() {
+		log.Error(ctx, map[string]interface{}{
+			"categories_id": categoryID,
+		}, "workitemtypes of category not found")
+		return nil, errors.NewNotFoundError("work item type category", categoryID.String())
+	}
+	if err := db.Error; err != nil {
+		return nil, errors.NewInternalError(db.Error.Error())
+	}
+	return relationship, nil
+}
+
+// LoadCategoryFromDB returns category for the given id
+// This is needed to check if a category is present in db or not.
+func (m *GormRepository) LoadCategoryFromDB(ctx context.Context, id uuid.UUID) (*Category, error) {
+	log.Logger().Infoln("Loading category", id)
+	res := Category{}
+	db := m.db.Model(&res).Where("id=?", id).First(&res)
+	if db.RecordNotFound() {
+		log.Error(ctx, map[string]interface{}{
+			"category_id": id,
+		}, "category not found")
+		return nil, errors.NewNotFoundError("category", id.String())
+	}
+	if err := db.Error; err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"categoryID": id,
```

Comment:
add the error as an additional key/value to the map `err`: err.Error(),

Also, remove it from the message in `category retrieval error`, err.Error())`